src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2366,7 +2365,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         if (!isPastTyper && meth.isPrimaryConstructor) {
           for (vparams <- ddef.vparamss; vd <- vparams) {
             if (vd.mods.isParamAccessor) {
-              vd.symbol setAnnotations (vd.symbol.annotations filter AnnotationInfo.mkFilter(ParamTargetClass, defaultRetention = true))
+              val sym = vd.symbol
+              sym.setAnnotations(sym.annotations.filter(AnnotationInfo.mkFilter(ParamTargetClass, defaultRetention = true)))
```

Comment:
Sacrificing your red vs green metric for the sake of your naming principles! I can respect that.